java-frontend/src/main/java/org/sonar/java/se/checks/NullDereferenceCheck.java

```diff
@@ -43,26 +43,25 @@
 
   @Override
   public ProgramState checkPreStatement(CheckerContext context, Tree syntaxNode) {
-    SymbolicValue currentVal = context.getState().peekValue();
-    if (currentVal == null) {
+    if (context.getState().stackIsEmpty()) {
```

Comment:
Why not simply create a local method `stackIsEmpty(ProgramState state)` doing the null-check?
As it's the only place you need it, creating a method in ProgramState seems over-engineering to me.